update test distributed

tktrungna opened the pull request:
Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62838
* #62797
* __->__ #62796
* #62795
* #62781

Differential Revision: [D30193142](https://our.internmc.facebook.com/intern/diff/D30193142)

tktrungna committed:
update test distributed

[ghstack-poisoned]

tktrungna committed:
Update on "update test distributed"




[ghstack-poisoned]

tktrungna committed:
Update on "update test distributed"




[ghstack-poisoned]

tktrungna committed:
Update on "update test distributed"




[ghstack-poisoned]

tktrungna committed:
Update on "update test distributed"




[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30193142).

tktrungna committed:
Update on "update test distributed"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30193142](https://our.internmc.facebook.com/intern/diff/D30193142)

[ghstack-poisoned]

tktrungna committed:
Update on "update test distributed"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30193142](https://our.internmc.facebook.com/intern/diff/D30193142)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30193142).

tktrungna committed:
Update on "update test distributed"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30193142](https://our.internmc.facebook.com/intern/diff/D30193142)

[ghstack-poisoned]

tktrungna committed:
Update on "update test distributed"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30193142](https://our.internmc.facebook.com/intern/diff/D30193142)

[ghstack-poisoned]

zhouzhuojie commented on test/cpp/c10d/CMakeLists.txt:
```diff
@@ -39,6 +49,11 @@ if(USE_CUDA)
+    if(INSTALL_TEST)
+      install(TARGETS ProcessGroupNCCLTest DESTINATION bin)
+      install(TARGETS ProcessGroupNCCLErrorsTest DESTINATION bin)
+      install(TARGETS c10d_cuda_test DESTINATION lib)
```
curious why this one is not `bin` but `lib`

tktrungna replied:
I am a cmake beginner so not sure if this is a correct solution, but basically `ProcessGroupNCCLTest` needs `libc10d_cuda_test.so` library. 
* Currently, `ProcessGroupNCCLTest` is built and put at `build/bin`, `libc10d_cuda_test.so` is compiled and put at `build/lib`
* In this PR, I've added installing step, which copies `ProcessGroupNCCLTest` to `{env}/site-packages/torch/bin`, and `libc10d_cuda_test.so` to `{env}/site-packages/torch/lib` (again would have to reluctantly add sym link to make the tests pass)

walterddr replied:
i guess the name is confusing it is not exactly a test but a test util library. 
but curious where was INSTALL_TEST defined in the first place? was it in another PR?

tktrungna replied:
`INSTALL_TEST` must be an option of the top-level CMakeLists.txt defined here https://github.com/pytorch/pytorch/blob/master/CMakeLists.txt#L174

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30193142).

tktrungna committed:
Update on "update test distributed"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30193142](https://our.internmc.facebook.com/intern/diff/D30193142)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30193142).
